### PR TITLE
tacit iptables support for Debian 9

### LINKED ISSUE #201: Cannot apply default configuration in Debian 9
### Steps to Reproduce: Create a recipe that uses `firewall 'default'` with attribute:
```
default['firewall']['ubuntu_iptables'] = true
```

### Actual Result:
```
I, [2018-12-17T14:55:12.425141 #7031]  INFO -- default-debian-9: [2018-12-17T22:55:10+00:00] FATAL: Mixlib::ShellOut::ShellCommandFailed: firewall[default] (machete.kvm::default line 25) had an error: Mixlib::ShellOut::ShellCommandFailed: service[iptables-persistent] (/tmp/kitchen/cache/cookbooks/firewall/libraries/provider_firewall_iptables_ubuntu1404.rb line 177) had an error: Mixlib::ShellOut::ShellCommandFailed: Expected process to exit with [0], but received '1'
I, [2018-12-17T14:55:12.425210 #7031]  INFO -- default-debian-9: ---- Begin output of /bin/systemctl --system enable iptables-persistent ----
I, [2018-12-17T14:55:12.425248 #7031]  INFO -- default-debian-9: STDOUT: 
I, [2018-12-17T14:55:12.425279 #7031]  INFO -- default-debian-9: STDERR: Failed to enable unit: File iptables-persistent.service: No such file or directory
I, [2018-12-17T14:55:12.425309 #7031]  INFO -- default-debian-9: ---- End output of /bin/systemctl --system enable iptables-persistent ----
I, [2018-12-17T14:55:12.425338 #7031]  INFO -- default-debian-9: Ran /bin/systemctl --system enable iptables-persistent returned 1
E, [2018-12-17T14:55:12.444482 #7031] ERROR -- default-debian-9: Converge failed on instance <default-debian-9>.
```

This may be related to #193 but I can't tell because they didn't provide a stack trace. On that note, there is a `netfilter-persistent` provider in [the libraries](https://github.com/chef-cookbooks/firewall/tree/master/libraries) but for some reason it is using the wrong one.

### PR DESCRIPTION
**Pull request by chaim1221**
### Description  

check to see if we are in fact on Debian before checking the OS version, to avoid provider thinking `> 14.04` applies to Debian  

### Issues Resolved  

resolves #201  

### Check List  

- [x] All tests pass. See <https://github.com/chef-cookbooks/community_cookbook_documentation/blob/master/TESTING.MD>  
- [x] New functionality includes testing . 
- [x] New functionality has been documented in the README if applicable  
- [x] All commits have been signed for the Developer Certificate of Origin. See <https://github.com/chef-cookbooks/community_cookbook_documentation/blob/master/CONTRIBUTING.MD>   

Notes:  

1. There was a test failure before I started (Debian 8, UFW should be running, fixed); creating the the rulesfile resulted in an unreadable line because of the following:

```
firewall_rule 'ufw raw test' do
  raw 'limit 23/tcp'
  only_if { %w(ubuntu debian).include?(node['platform_family']) && !node['firewall']['ubuntu_iptables'] }
end
```

... I did not fix the issue for Debian 9 as I have no interest in supporting Uncomplicated Firewall.

2. The Windows 2012 R2 image could not be found, at least for my provider (virtualbox).

### CODE REVIEW

**Comment by chaim1221** [comment]
I see the style violations. I will deal with them tomorrow.

**Comment by chaim1221** [comment]
There's also a larger problem with this, which is that I assumed the `default` test suite was only testing `ufw`. I don't know why I assumed that.

**Comment by chaim1221** [comment]
Above comments are resolved.

File: README.md [hunk 1]
```diff
@@ -16,14 +16,15 @@ depends 'firewall', '< 2.0'
 ```
 
 ### Supported firewalls and platforms
-* UFW - Ubuntu, Debian
+* UFW - Ubuntu, Debian (except 9)
```
Comment (chaim1221) [inline README.md]: Technically, it should work, and the tests pass. This may be unnecessary.

**Comment by martinb3** [comment]
Thanks @chaim1221! I've released this as v2.7.0.

**Pull request merged**